build.gradle

```diff
@@ -234,40 +232,41 @@ task ktlint(type: JavaExec, group: "verification") {
 check.dependsOn ktlint
 check.dependsOn verifyPlugin
 
-def publishedProjects = subprojects.findAll { !it.name.contains("test")}
-
 task coverageReport(type: JacocoReport) {
     executionData fileTree(project.rootDir.absolutePath).include("**/build/jacoco/*.exec")
 
-    getAdditionalSourceDirs().from(publishedProjects.sourceSets.main.java.srcDirs)
-    getSourceDirectories().from(publishedProjects.sourceSets.main.java.srcDirs)
-    getClassDirectories().from(publishedProjects.sourceSets.main.output.classesDirs)
+    getAdditionalSourceDirs().from(subprojects.sourceSets.main.java.srcDirs)
+    getSourceDirectories().from(subprojects.sourceSets.main.java.srcDirs)
+    getClassDirectories().from(subprojects.sourceSets.main.output.classesDirs)
 
     reports {
         html.enabled true
         xml.enabled true
     }
 }
-publishedProjects.forEach {
+subprojects.forEach {
     coverageReport.mustRunAfter(it.tasks.withType(Test))
 }
 check.dependsOn coverageReport
 
 // Workaround for runIde being defined in multiple projects, if we request the root project runIde, "alias" it to
 // community edition
 if (gradle.startParameter.taskNames.contains("runIde")) {
-    println("Top level runIde selected, excluding sub-projects' runIde")
-    gradle.taskGraph.whenReady { graph ->
-        graph.allTasks.forEach {
-            if (gradle.startParameter.systemPropertiesArgs.getOrDefault("exec.args", "").contains("guitest")) {
-                if (it.name == "runIde" && it.project != project(':jetbrains-core-gui')) {
-                    it.enabled = false
-                }
-            } else {
-                if (it.name == "runIde" &&
+    // Only disable this if running from root project
```

Comment:
Uh that is not the UI I was expecting after making that comment, commenting on the most recent version of the PR:
nit: this comment is now backwards

Reply:
Nevermind, the comment is correct, the logic is just kind of weird